### PR TITLE
Cleanup macos common conda installation

### PR DESCRIPTION
**Pull request by huydhn**
The conda dependencies have all been installed for `_mac-test` in https://github.com/pytorch/pytorch/pull/87541.  I missed the same step for `_mac-build` and `_mac-test-mps` workflows, so both are also updated here. Note that arm64 is cross-compiled from x86, so the env file needs to be set explicitly in that case

After this one, I have a WIP PR to consolidate macos pip dependencies next

### CODE REVIEW

File: .jenkins/pytorch/macos-common.sh [hunk 1]
```diff
@@ -7,52 +7,6 @@ source "$(dirname "${BASH_SOURCE[0]}")/common.sh"
...
-# The torch.hub tests make requests to GitHub.
-#
-# The certifi package from conda-forge is new enough to make the
-# following error disappear (included for future reference):
-#
-# > ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED]
-# > certificate verify failed: unable to get local issuer certificate
-# > (_ssl.c:1056)
-#
-retry conda install -y -c conda-forge certifi wheel=0.36.2
```
Comment (ZainRizvi) [inline .jenkins/pytorch/macos-common.sh:48]: I don't see certifiwheel in https://github.com/huydhn/pytorch/blob/master/.github/requirements/conda-env-macOS-X64. 

Is this package not needed anymore?

**Review by ZainRizvi — CHANGES_REQUESTED** [review]

File: .jenkins/pytorch/macos-common.sh (hunk 1 above)
Comment (huydhn) [reply .jenkins/pytorch/macos-common.sh:48]: Oh,they are certifi and wheel packages. The first is included in these conda env files whle the latter has already been set in https://github.com/pytorch/test-infra/blob/main/.github/actions/setup-miniconda/action.yml#L122 across all conda envs

File: .github/requirements/conda-env-macOS-ARM64 [hunk 2]
```diff
@@ -8,9 +8,10 @@ dataclasses=0.8
 pip=22.2.2
 six=1.16.0
 pillow=9.2.0
-libuv=1.39.0
+libuv=1.40.0
```
Comment (huydhn) [inline .github/requirements/conda-env-macOS-ARM64:11]: I realize that this version is wrong and should be `1.40.0`, same as the one from `conda-env-macOS-X64`.  Using 1.39.0 here causes a conda dependency conflict

File: .github/requirements/conda-env-macOS-ARM64 [hunk 3]
```diff
@@ -8,9 +8,10 @@ dataclasses=0.8
...
 pkg-config=0.29.2
 wheel=0.37.1
+expecttest=0.1.3
```
Comment (huydhn) [inline .github/requirements/conda-env-macOS-ARM64:14]: This one is copied from `_mac_test_mps`

**Review by ZainRizvi — APPROVED** [review]

**Pull request closed without merging**